native/cocos/core/Root.cpp

```diff
@@ -62,8 +62,9 @@ Root::Root(gfx::Device *device)
 }
 
 Root::~Root() {
-    instance = nullptr;
+    destroy();
     CC_SAFE_DELETE(_eventProcessor);
+    instance = nullptr;
```

Comment:
Will `~Root` be invoked?

Reply:
Yep, it will be invoked when invoking `ScriptEngine::cleanup()`.